csrc/kernel_cache.cpp

```diff
@@ -724,7 +726,12 @@ FusionKernelRuntime* FusionExecutorCache::getKernelRuntimeFor(
     }
     FusionGuard fg(conc_fusion.get());
     kernel_runtimes.emplace_back(std::make_unique<FusionKernelRuntime>(
-        std::move(conc_fusion), args, forced_index_type));
+        std::move(conc_fusion),
+        args,
+        forced_index_type,
+        fusion_id_,
+        args.getDeviceIndex(),
+        kernel_runtimes.size()));
```

Comment:
Note there is another level here that's not being indexed I believe. `kernel_runtimes_` is an unordered_map from (device, conc_info) pairs to vectors of runtimes: a single concretization can have multiple runtimes since there might be a number of different input sizes that lead to different vectorization strategies, for example. So you'd need to index the concretization info, but that is not currently ordered.